homeassistant/components/viam/manager.py

```diff
@@ -0,0 +1,114 @@
+"""Manage Viam client connection."""
+
+import base64
+from typing import Any
+
+from PIL import Image
+from viam.app.app_client import RobotPart
+from viam.app.viam_client import ViamClient
+from viam.robot.client import RobotClient
+from viam.rpc.dial import Credentials, DialOptions
+from viam.services.vision.client import RawImage
+
+from homeassistant.components import camera
+from homeassistant.const import CONF_ADDRESS
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ServiceValidationError
+
+from .const import (
+    CONF_API_ID,
+    CONF_CREDENTIAL_TYPE,
+    CONF_ROBOT_ID,
+    CONF_SECRET,
+    CRED_TYPE_LOCATION_SECRET,
+    DOMAIN,
+)
+
+
+def _fetch_image(filepath: str | None):
+    if filepath is None:
+        return None
+    return Image.open(filepath)
+
+
+class ViamManager:
+    """Manage Viam client and entry data."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        viam: ViamClient,
+        entry_id: str,
+        data: dict[str, Any],
+    ) -> None:
+        """Store initialized client and user input data."""
+        self.hass = hass
+        self.viam = viam
+        self.data = data
+        self.entry_id = entry_id
+
+    def unload(self) -> None:
+        """Clean up any open clients."""
+        self.viam.close()
+
+    def encode_image(self, image: Image.Image | RawImage):
```

Comment:
Instead of a method, move this and `get_image` to a separate module `util.py` for example

Reply:
Since these are only used by `services.py`, I moved them there.